### PR TITLE
Replace usage of sysout with slf4j

### PR DESCRIPTION
**Pull request by KengoTODA**
This PR replaces usage of `System.out` and `SystemProperties.getBoolean` with SLF4J partially.

### CODE REVIEW

File: spotbugs/src/main/java/edu/umd/cs/findbugs/CategoryFilteringBugReporter.java [hunk 1]
```diff
@@ -43,9 +46,7 @@ public void reportBug(@Nonnull BugInstance bugInstance) {
         if (categorySet.contains(category)) {
             getDelegate().reportBug(bugInstance);
         } else {
-            if (DEBUG) {
-                System.out.println("CategoryFilteringBugReporter: filtered due to category " + category);
-            }
+            LOG.debug("CategoryFilteringBugReporter: filtered due to category " + category);
```
Comment (KengoTODA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/CategoryFilteringBugReporter.java]:
```suggestion
            LOG.debug("CategoryFilteringBugReporter: filtered due to category {}", category);
```

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/bcp/PatternElement.java [hunk 2]
```diff
@@ -211,9 +212,7 @@ protected static BindingSet addOrCheckDefinition(String varName, Variable variab
             bindingSet = new BindingSet(new Binding(varName, variable), bindingSet);
         } else {
             if (!existingVariable.sameAs(variable)) {
-                if (DEBUG) {
-                    System.out.println("\tConflicting variable " + varName + ": " + variable + " != " + existingVariable);
-                }
+                LOG.debug("\tConflicting variable {}: {} != {}", varName, variable, existingVariable);
```
Comment (wreulicke) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/ba/bcp/PatternElement.java]: not needs tabs?

Comment (KengoTODA) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/ba/bcp/PatternElement.java]: I guess here tab is used to group multiple log entries like `console.group()`?

Comment (KengoTODA) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/ba/bcp/PatternElement.java]: 29b7fe000c6e4e6370dce812e60f834379e0502e introduced this tab. I cannot find its intention...

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/bcp/PatternMatcher.java [hunk 3]
```diff
@@ -448,11 +450,11 @@ private MatchResult matchLocation(Location location) throws DataflowAnalysisExce
     }
 
     private void debug(String s) {
-        if (!DEBUG) {
+        if (!LOG.isDebugEnabled()) {
             throw new IllegalStateException("Only call if DEBUG is true");
         }
-        System.out.print("                                            ".substring(0, depth));
-        System.out.println(s);
+        String indent = "                                            ".substring(0, depth);
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/ba/bcp/PatternMatcher.java]: This really should be replaced by `StringUtils.repeat` or similar functionality.

File: spotbugs/src/main/java/edu/umd/cs/findbugs/PluginLoader.java [hunk 4]
```diff
@@ -926,30 +902,22 @@ private void loadPluginComponents()
...
+                        LOG.debug("rejected abbrev '{}' for category {}: {}", abbrev, key, bc.getAbbrev());
                     }
                 } catch (PluginException pe) {
-                    if (DEBUG)
-                    {
-                        System.out.println("missing Abbreviation for category " + key + "/" + shortDesc);
-                        // do nothing else -- Abbreviation is required, but handle
-                        // its omission gracefully
-                    }
+                    System.out.println("missing Abbreviation for category " + key + "/" + shortDesc);
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/PluginLoader.java]: Still uses `System.out`

File: spotbugs/src/main/java/edu/umd/cs/findbugs/AbstractBugReporter.java [hunk 5]
```diff
@@ -49,9 +51,7 @@
  * BugReporter objects.
  */
 public abstract class AbstractBugReporter implements BugReporter {
-    private static final boolean DEBUG = SystemProperties.getBoolean("abreporter.debug");
-
-    private static final boolean DEBUG_MISSING_CLASSES = SystemProperties.getBoolean("findbugs.debug.missingclasses");
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractBugReporter.class);
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/AbstractBugReporter.java]: To make copy-pasting easier, you can use `LoggerFactory.getLogger(MethodHandles.lookup().lookupClass())`

**Pull request merged**